documentation/src/main/asciidoc/reference/introduction.adoc

```diff
@@ -1006,89 +1034,82 @@ currentContext.runOnContext( event -> {
+  }
+
+}
+----
+
+One way to register this implementation is to configure Hibernate programmatically,
+for example:
 
 [source, JAVA, indent=0]
 ----
-final Vertx myVertx = ...
 Configuration configuration = new Configuration();
 StandardServiceRegistryBuilder builder = new ReactiveServiceRegistryBuilder()
-        .addService( VertxInstance.class, (VertxInstance) () -> myVertx )
+        .addService( VertxInstance.class, new MyVertx() )
         .applySettings( configuration.getProperties() );
 StandardServiceRegistry registry = builder.build();
 SessionFactory sessionFactory = configuration.buildSessionFactory( registry );
 ----
 
-Or, you can use the `ServiceContributor` interface and register your implementation of the
-`VertxInstance` interface:
+Alternatively, you could implement the `ServiceContributor` interface.
 
-1. Add a text file named `org.hibernate.service.spi.ServiceContributor` under 
-+
-[source, JAVA, indent=0]
-----
-/META-INF/services/
-----
-
-2. The text file will contain the path to the class implementing the `ServiceContributor` interface: 
-+
-[source, JAVA, indent=0]
-----
-org.myproject.MyServiceContributor
-----
-+
-This is an example of what `MyVertxProvider` and `MyServiceContributor` might look like:
-+
 [source, JAVA, indent=0]
 ----
 public class MyServiceContributor implements ServiceContributor {
-
   @Override
   public void contribute(StandardServiceRegistryBuilder serviceRegistryBuilder) {
     serviceRegistryBuilder.addService( VertxInstance.class, new MyVertxProvider() );
   }
-
-}
-----
-+
-[source, JAVA, indent=0]
-----
-public class MyVertxProvider implements VertxInstance {
-
-  private final Vertx vertx;
-
-  public MyVertxProvider() {
-    this.vertx = Vertx.vertx();
-  }
-
-  @Override
-  public Vertx getVertx() {
-    return vertx;
-  }
-
 }
 ----
 
-You can access the `ServiceRegistry` from the `Stage.SessionFactory` or
-`Mutiny.SessionFactory` using the `org.hibernate.reactive.common.spi.Implementor`
-interface:
+To register this `ServiceContributor`, add a text file named
+`org.hibernate.service.spi.ServiceContributor` to `/META-INF/services/`.
 
 [source, JAVA, indent=0]
 ----
-Mutiny.SessionFactory sessionFactory = ...
-VertxInstance vertxInstance = ((Implementor) sessionFactory)
-					.getServiceRegistry()
- 					.getService( VertxInstance.class );
-Vertx vertx = vertxInstance.getVertx();
+org.myproject.MyServiceContributor
 ----
 
-The same will work for `Stage.SessionFactory`.
+//If you need to, you can access the `ServiceRegistry` from the `Stage.SessionFactory`
+//or `Mutiny.SessionFactory` using the `org.hibernate.reactive.common.spi.Implementor`
+//interface:
+//
+//[source, JAVA, indent=0]
+//----
+//Mutiny.SessionFactory sessionFactory = ...
+//VertxInstance vertxInstance = ((Implementor) sessionFactory)
+//					.getServiceRegistry()
+// 					.getService( VertxInstance.class );
+//Vertx vertx = vertxInstance.getVertx();
+//----
+//
+//The same will work for `Stage.SessionFactory`.
```

Comment:
Are you planning to add this back somewhere? Maybe in a separate section about the service registry?

Reply:
Well I had two issues with this section:

1. It seemed to be describing some very technical interfaces that aren't of use to the casual user, and didn't seem to belong in an introductory guide.
2. It didn't explain why I might need it.

But, if you can give a good motivation for it being there, then I'll to add that motivation to the text and uncomment this passage.

Reply:
I guess is not useful for most users but accessing the vertx instance when it's created by Hibernate Reactive might be helpful.

I don't have a use case though and it's an advance feature, so I don't mind to omit it for now.